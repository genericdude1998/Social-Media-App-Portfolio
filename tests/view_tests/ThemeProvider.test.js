import React, { Children } from 'react';
import { shallow } from 'enzyme';
import ConnectedThemeProvider, { ThemeProvider } from '../../src/view/theme/ThemeProvider';
import { mockTheme } from '../mockValues';

let wrapper;
const mockOnToggleTheme = jest.fn();

wrapper = shallow(
    <ConnectedThemeProvider.WrappedComponent
        theme={mockTheme}
        onToggleTheme={mockOnToggleTheme}  
    />
);

console.log(wrapper.debug());

describe('ThemeProvider', () => {
    it('should pass down theme variable to ThemeContextProvider and onToggleTheme', () => {
        expect(wrapper.prop('value').theme).toBe(mockTheme);
        expect(wrapper.prop('value').onToggleTheme).toBe(mockOnToggleTheme);
    });
    it('should matchSnapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});