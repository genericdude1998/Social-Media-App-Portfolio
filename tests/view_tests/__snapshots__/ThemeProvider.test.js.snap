// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider should matchSnapshot 1`] = `
<ContextProvider
  value={
    Object {
      "onToggleTheme": [MockFunction],
      "theme": "mock_theme",
    }
  }
/>
`;
